protocol/transfer-precompile.md

```diff
@@ -0,0 +1,83 @@
+# Purpose
+
+Since its inception, the Celo blockchain provides a feature called "[Token
+Duality](https://specs.celo.org/token_duality.html)", which allows the native
+token to be used both in the usual ETH-like way and as an ERC-20 token.
+
+We would like to bring this feature to more OP-Stack based chains and bring the
+Celo L2 closer to vanilla OP-stack by adding the underlying transfer precompile to
+op-geth.
+
+# Summary
+
+Add a precompile to op-geth that allows transferring native tokens.
+This precompile can only be called by a single privileged contract.
+
+# Problem Statement + Context
+
+Many users and contracts have to deal with both ERC20 tokens and native tokens.
+Since they can't be accessed by the same interface, they either have to be
+handled as two different cases or be explicitly wrapped by an ERC-20 token
+wrapping contract.
+Wrapping tokens is not an ideal solution either, since the wrapped token is
+of limited use (e.g. can't be used to pay gas) and wrapping/unwrapping is
+annoying and incurs additional gas charges.
+
+This can be solved by having a contract that provides an ERC-20 interface
```

Comment:
So the main value prop here is that the unwrap/wrap steps are saved. In a world with smart contract wallets, users can do multicalls and save needing to send multiple transactions, but this is still cheaper as there is no wrap/unwrap step necessary

Reply:
Yes, but there is plenty of benefit before everyone uses smart contract wallets and every app has implemented perfect wrap/unwrap handling (Should tokens stay wrapped after the action to save gas? Should existing wrapped tokens be used? Even if they are not enough so that you have to wrap anyway? Do you want to bulk wrap to save gas on the next actions? Should the wallet UI show wrapped tokens separately from unwrapped tokens?).

There is also the argument that it makes the app developers lives easier since they don't have to implement all these wrap/unwrap cases. But since most apps are deployed across many chains, most devs will have to do that anyway.